### PR TITLE
Add end to end example docs to the github pages menu

### PR DESCRIPTION
**Pull request by bpkroth**

### CODE REVIEW

File: website/content/menu/index.md [hunk 1]
```diff
@@ -10,14 +10,17 @@ headless = true
...
   - [Bayesian Optimization]({{< relref "/notebooks/BayesianOptimization" >}})
   - [Smart Cache Optimization]({{< relref "/notebooks/SmartCacheOptimization" >}})
 
-
 - API Documentation
   - [Python](/MLOS/python_api/)
 
+- [E2E Examples]({{< relref "/source/Examples" >}})
+  - [Smart Cache]({{< relref "/source/Examples/SmartCache" >}})
+  - [Smart Shared Channel]({{< relref "/source/Examples/SmartSharedChannel" >}})
```
Comment (amueller) [inline website/content/menu/index.md:22]: Is this one ready for the students? I haven't tested it yet.

Comment (amueller) [reply website/content/menu/index.md:22]: Hm... https://microsoft.github.io/MLOS/source/Examples/SmartSharedChannel/

Comment (bpkroth) [reply website/content/menu/index.md:22]: It's not as well documented, and isn't really meant as an educational tool, but they could use it.

Comment (amueller) [reply website/content/menu/index.md:22]: I think I would remove it for now if you don't mind.

Comment (bpkroth) [reply website/content/menu/index.md]: Done.

**Review by amueller — APPROVED** [review]

**Pull request merged**